tests/PHPStan/Analyser/data/falsy-isset.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace FalsyIsset;
+
+use function PHPStan\Testing\assertType;
+use function PHPStan\Testing\assertVariableCertainty;
+use PHPStan\TrinaryLogic;
+
+class HelloWorld
+{
+	public function undefinedVar(): void
+	{
+		if (isset($a['bar'])) {
+			assertType("*ERROR*", $a);
+		} else {
+			assertType("*ERROR*", $a);
+		}
+	}
+
+	public function maybeCertainNull(): void
+	{
+		if (rand() % 2) {
+			$a = ['bar' => null];
+			if (rand() % 3) {
+				$a = ['bar' => 'hello'];
+			}
+		}
+		assertVariableCertainty(TrinaryLogic::createMaybe(), $a);
+
+		assertType("array{bar?: null}|array{bar?: 'hello'}", $a);
+		if (isset($a['bar'])) {
+			assertVariableCertainty(TrinaryLogic::createYes(), $a);
+			assertType("array{bar: 'hello'}", $a);
+			$a['bar'] = 1;
+			assertType("array{bar: 1}", $a);
+		} else {
+			assertVariableCertainty(TrinaryLogic::createMaybe(), $a);
+			assertType('array{bar?: null}', $a);
+		}
+
+		assertVariableCertainty(TrinaryLogic::createMaybe(), $a);
+		assertType("array{bar?: null}|array{bar: 1}", $a);
+	}
+
+	public function maybeCertainNonNull(): void
+	{
+		if (rand() % 2) {
+			$a = ['bar' => 'world'];
+			if (rand() % 3) {
+				$a = ['bar' => 'hello'];
+			}
+		}
+		assertVariableCertainty(TrinaryLogic::createMaybe(), $a);
+
+		assertType("array{bar?: 'hello'}|array{bar?: 'world'}", $a);
```

Comment:
atm this errors with `"array{bar: 'hello'}|array{bar: 'world'}"` - even before this PR.
not sure whether thats a separate pre existing problem or my expectation here is wrong